### PR TITLE
[JBIDE-23505] - CDK v3 server adapter based on minishift

### PR DESCRIPTION
**Pull request by jeffmaury**
- Fix for Windows platforms

Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

**Comment by jeffmaury** [comment]
@robstryker can you review I tested on Win10

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ServiceManagerUtility.java [hunk 1]
```diff
@@ -17,17 +17,17 @@
 	public static HashMap<String, String> parseLines(String[] lines) {
 		HashMap<String, String> adbEnv = new HashMap<>();
 
-		String setEnvWin = "setx ";
```
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ServiceManagerUtility.java]: I'm very confused why you're making this patch. First, this class is used for both cdk 2.x and 3.x, so it needs to work for both. I know for a fact cdk 2.x on windows with certain plugin versions will still return the setx syntax, so we cannot remove parsing ability for that. 

Do you have any confirmation it has switched to SET from setx?  Is this based on what shell you are using? 

I think this patch will likely break things.

Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ServiceManagerUtility.java]: I checked that on my Win10 box CDKv2 vagrant service-manager env returns the following:

    > vagrant service-manager env                                                                                                  
                                                                                                                               
    # docker env:                                                                                                                  
    # Set the following environment variables to enable access to the                                                              
    # docker daemon running inside of the vagrant virtual machine:                                                                 
    export DOCKER_HOST=tcp://10.1.2.2:2376                                                                                         
    export DOCKER_CERT_PATH='C:\Users\JeffMAURY\CDK\2.3\components\rhel\rhel-    ose\.vagrant\machines\default\virtualbox\docker\'     
    export DOCKER_TLS_VERIFY=1                                                                                                     
    export DOCKER_API_VERSION=1.22                                                                                                 
                                                                                                                               
                                                                                                                               
    # openshift env:                                                                                                               
    # You can access the OpenShift console on: https://10.1.2.2:8443/console                                                       
    # To use OpenShift CLI, run: oc login https://10.1.2.2:8443                                                                    
    export OPENSHIFT_URL=https://10.1.2.2:8443                                                                                     
    export OPENSHIFT_WEB_CONSOLE=https://10.1.2.2:8443/console                                                                     
    export DOCKER_REGISTRY=hub.openshift.rhel-cdk.10.1.2.2.xip.io                                                                  
                                                                                                                               
    # run following command to configure your shell:                                                                               
    # eval "$(VAGRANT_NO_COLOR=1 vagrant service-manager env | tr -d '\r')"                                                        
                                                                                                                               
At least, the CDKv3 on Windows is using the SET notation:

    > minishift docker-env                                             
    SET DOCKER_TLS_VERIFY=1                                            
    SET DOCKER_HOST=tcp://192.168.99.104:2376                          
    SET DOCKER_CERT_PATH=C:\Users\JeffMAURY\.minishift\certs           
    SET DOCKER_API_VERSION=1.23                                        
    REM Run this command to configure your shell:                      
    REM     @FOR /f "tokens=*" %i IN ('minishift docker-env') DO @%i   

so the current code does not on Windows. I'm gonna check if I can make CDKv2 issue setx statements on Windows

**Comment by adietish** [comment]
@robstryker @jeffmaury can we please fix the simple issues reported by sonar? 
* 3 missing @Override that are reported by sonar? 
* There's also a return value of HashMap where the interface Map would be preferable.

**Comment by robstryker** [comment]
* f6097f3 - (HEAD -> master, origin/master, origin/HEAD, JBIDE-23505) [JBIDE-23505] - cleanup and unit test (58 seconds ago) <Rob Stryker>
* 27269b3 - [JBIDE-23505] - CDK v3 server adapter based on minishift (23 minutes ago) <Jeff MAURY>


I've made changes / additions to @jeffmaury 's code and added a test. I pushed it withotu approval, and don't feel bad about it ;)

**Pull request closed without merging**